src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +272,32 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary when of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummaryDoc = getInlineSummarySentence(ast);
+        if (inlineSummaryDoc.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            if (!inlineSummaryDoc.contains(period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            final String inlineFirstSentence = getInlineFirstSentence(ast);
+            final int endOfInlineSentence = inlineFirstSentence.lastIndexOf(period);
+            if (endOfInlineSentence != -1
+                    && (containsForbiddenFragment(inlineFirstSentence.substring(0,
+                    endOfInlineSentence))
```

Comment:
this 5 line above should be new method, something like
```
if (isFirstSentenceAllowed(ast)
     || ....)
```

Reply:
Done